src/libs/actions/Workflow.ts

```diff
@@ -0,0 +1,196 @@
+import type {OnyxCollection, OnyxUpdate} from 'react-native-onyx';
+import Onyx from 'react-native-onyx';
+import * as API from '@libs/API';
+import type {CreateWorkspaceApprovalParams, RemoveWorkspaceApprovalParams, UpdateWorkspaceApprovalParams} from '@libs/API/parameters';
+import {WRITE_COMMANDS} from '@libs/API/types';
+import * as NetworkStore from '@libs/Network/NetworkStore';
+import {convertApprovalWorkflowToPolicyEmployees} from '@libs/WorkflowUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {ApprovalWorkflow, Policy} from '@src/types/onyx';
+import type {Approver, Member} from '@src/types/onyx/ApprovalWorkflow';
+
+let currentApprovalWorkflow: ApprovalWorkflow | undefined;
+Onyx.connect({
+    key: ONYXKEYS.APPROVAL_WORKFLOW,
+    callback: (approvalWorkflow) => {
+        currentApprovalWorkflow = approvalWorkflow;
+    },
+});
+
+let allPolicies: OnyxCollection<Policy>;
+Onyx.connect({
+    key: ONYXKEYS.COLLECTION.POLICY,
+    waitForCollectionCallback: true,
+    callback: (value) => (allPolicies = value),
+});
+
+function createApprovalWorkflow(policyID: string, approvalWorkflow: ApprovalWorkflow) {
+    const authToken = NetworkStore.getAuthToken();
```

Comment:
`getAuthToken()` is a new method for me. I think this file should use `Onyx.connect()` and grab the authToken from the SESSION onyx key. Again, maybe there was a good reason for this change that I didn't know about. 

Typically, people have proposed using singletons like this because of a perception that having lots of `Onyx.connect()` for the same key is bad for performance. Yet, I haven't ever seen evidence of this problem and I always direct people to keep using `Onyx.connect()`.

The downsides to singletons is that they end up being used in react components to get Onyx data, which results in the component not being "reactive" to data changes at all. So, if we use singletons, we must be sure to only use them in lib code, and only when there is a good and clear reason to use them.

Reply:
You can see in https://github.com/Expensify/App/issues/27262 the big effort that I had to go through to fix a lot of those bad patterns.

Reply:
Thanks for the context, I looked for authToken in other actions and saw this pattern that's I went for it. Changing it to Onyx.connect 😄